### PR TITLE
OCPBUGS-1904: Only deploy VolumeSnapshotClass when CRD exists

### PR DESCRIPTION
**Pull request by bertinatto**
When the cluster-csi-snapshot-controller-operator is not deployed, (through the CSISnapshot capability) the VolumeSnapshotClass CRD is not created at all.

In that case, it doesn't make sense to deploy the CR, otherwise the operator gets degraded.

CC @openshift/storage

### CODE REVIEW

File: pkg/operator/volumesnapshotclasscontroller/volumesnapshotclasscontroller.go [hunk 1]
```diff
@@ -59,6 +64,13 @@ func (c *VolumeSnapshotClassController) sync(ctx context.Context, syncCtx factor
 		return nil
 	}
 
+	crdName := "volumesnapshotclasses.snapshot.storage.k8s.io"
+	_, err = c.apiExtClient.ApiextensionsV1().CustomResourceDefinitions().Get(ctx, crdName, metav1.GetOptions{})
```
Comment (jsafrane) [inline pkg/operator/volumesnapshotclasscontroller/volumesnapshotclasscontroller.go:68]: @bertinatto maybe I should have noticed it earlier, do our operators have GET permissions on CRD? They will silently stop deploying snapshot classes if not.

Comment (bertinatto) [reply pkg/operator/volumesnapshotclasscontroller/volumesnapshotclasscontroller.go:68]: @jsafrane yes: https://github.com/openshift/cluster-storage-operator/pull/323/files

Comment (jsafrane) [reply pkg/operator/volumesnapshotclasscontroller/volumesnapshotclasscontroller.go:68]: https://github.com/openshift/cluster-storage-operator/pull/323

**Pull request merged**